tests/sdn/persistentips.go

```diff
@@ -0,0 +1,133 @@
+package sdn
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	nadv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	kvirtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/checks"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libmigration"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = SIGDescribe("Persistent IPs", decorators.PersistentIPs, func() {
+
+	Context("with a VirtualMachine configured with a allowPersistentIPs nad", func() {
+		var (
+			vm  *kvirtv1.VirtualMachine
+			vmi *kvirtv1.VirtualMachineInstance
+			nad *nadv1.NetworkAttachmentDefinition
+		)
+		BeforeEach(func() {
+			Expect(checks.HasFeature(virtconfig.PersistentIPsGate)).To(BeTrue(), "should have persistent ips feature gate enabled")
+
+			nad = &nadv1.NetworkAttachmentDefinition{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "persistentips",
+					Namespace: testsuite.GetTestNamespace(nil),
+				},
+				Spec: nadv1.NetworkAttachmentDefinitionSpec{
+					Config: fmt.Sprintf(`
+{
+        "cniVersion": "0.3.0",
+        "name": "persistentips",
+        "type": "ovn-k8s-cni-overlay",
+        "topology": "layer2",
+        "subnets": "10.100.200.0/24",
+        "netAttachDefName": "%s/persistentips",
+        "allowPersistentIPs": true
+}
+`, testsuite.GetTestNamespace(nil)),
+				},
+			}
+			By("Create NetworkAttachmentDefinition")
+			_, err := kubevirt.Client().NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(testsuite.GetTestNamespace(nil)).Create(context.TODO(), nad, metav1.CreateOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			vm = libvmi.NewVirtualMachine(libvmifact.NewAlpine(
+				libvmi.WithNamespace(testsuite.GetTestNamespace(nil)),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding("multus")),
+				libvmi.WithNetwork(libvmi.MultusNetwork("multus", "persistentips")),
+			), libvmi.WithRunning())
+
+			By("Creating VM using the nad")
+			vm, err = kubevirt.Client().VirtualMachine(vm.Namespace).Create(context.Background(), vm, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Wait for VMI to be created")
+			Eventually(matcher.ThisVMIWith(vm.Namespace, vm.Name)).
+				WithPolling(time.Second).
+				WithTimeout(2 * time.Minute).
+				ShouldNot(BeNil())
```

Comment:
why have you changed this from `Should(Succeed())` ?

Reply:
The helper `ThisVMIWith` return nil if VMI is not found, so we should check the pointer instead of an error.

Reply:
Right, got "fooled" by the API. Should have checked. Thank you.